cmd/autoowners/main.go

```diff
@@ -294,18 +326,25 @@ func pullOwners(directory string, whitelist []string) ([]string, error) {
 
 	config := filepath.Join(operatorRoot, "config")
 	templates := filepath.Join(operatorRoot, "templates")
-	for _, orgRepo := range orgRepos {
-		logrus.WithField("orgRepo", fmt.Sprintf("%+v", *orgRepo)).Info("handling repo ...")
+	for i, orgRepo := range orgRepos {
+		logrus.WithField("index", i).
+			WithField("orgRepo", fmt.Sprintf("%s/%s", orgRepo.Organization, orgRepo.Repository)).
+			Info("handling repo ...")
 		err = orgRepo.getDirectories(config, templates)
 		if err != nil && !os.IsNotExist(err) {
 			return repos, err
 		}
 
 		err = orgRepo.getOwnersHTTP()
-		if err != nil && !os.IsNotExist(err) {
+		if err != nil {
+			// TODO we might need to handle errors from `yaml.Unmarshal` if OWNERS is not a valid yaml file
 			return repos, err
 		}
-
+		if !orgRepo.OwnersFileExists {
+			logrus.WithField("orgRepo", fmt.Sprintf("%s/%s", orgRepo.Organization, orgRepo.Repository)).
+				Warn("Ignore the repo with no OWNERS file in the upstream repo.")
+			continue
+		}
```

Comment:
I must admit I am not a great fan of this code structure where an external method (`pullOwners`) implements the main algorithm by calling a magic sequence of private methods of `orgRepo` which stores state...

But I'm aware that you basically inherited this, but I think we should eventually refactor this.

Reply:
Will work on this today.

Reply:
@petr-muller refactoring is done.
Please check if it gets any better:

    Refactor pullOwners function
    
    * No changes orgRepo instances.
       * Compute resolved OWNERS by a function
       * No method of orgRepo is defined via pointer
    
    * Compute modified folders by `git-status` command.